### PR TITLE
75 Show better error message in the console when unable to create the button

### PR DESCRIPTION
**Pull request by paOol**
https://github.com/PayButton/paybutton/issues/75

renders the component 
displays errors
disables button

### CODE REVIEW

File: paybutton/src/index.tsx [hunk 1]
```diff
@@ -104,7 +104,7 @@ function findAndRender <T>( className: string, Component: React.ComponentType<an
 
       if ( ! requiredProps.every( name => name in attributes ) ) {
         console.error( 'PayButton: missing required attribute: ' + JSON.stringify( requiredProps.filter( name => ! ( name in attributes ) ) ) );
-        return;
+        // return;
```
Comment (Klakurka) [inline paybutton/src/index.tsx:107]: Meant to be commented out still?

**Review by Klakurka — CHANGES_REQUESTED** [review]
What's the main functional difference with this one?

The goal was to make it clearer to the user what is wrong when they've made mistakes in the set up.

For example,

1. Can't find element ID on page (either missing or misspelt).
2. Garbage entered for any of the parameters.

Ideally, we'd say "Hey, X parameter was set incorrectly. Please check it." in the console.

They currently see an unhelpful message. Eg. here is what happens when it can't find the element ID of the div to create the button in:

image

**Comment by paOol** [comment]
The original flow with a missing "to" param was to not render anything and just have a console warning. 

Now the button renders (disabled) and the erorr msg is shown in the component as well as the console. 
Other errors can also be shown this way to make them more visible.


`can't find the element ID of the div to create the button in:`

I haven't tested this yet but can include it into this PR.

File: paybutton/src/index.tsx (hunk 1 above)
Comment (paOol) [reply paybutton/src/index.tsx:107]: yes,  for someone in the future to be able to see that and easily revert if needed. versus having to dig through git history.

**Comment by Klakurka** [comment]
Yeah, include that.

It's supposed to also be able to handle garbage inputs. Anytime there's something that would throw an error (because of a bad input), it would let them know which parameter is messed up.

**Comment by paOol** [comment]
testing should be done by building and creating via HTML or JS.

storybook is clunky to force bad values.

**Comment by Klakurka** [comment]
image

I get this when I view it on localhost:10001 and remove the `to` field.

**Comment by Klakurka** [comment]
A few things seem to be broken... not sure what the deal is:

 - 'Click to copy' isn't working for me.
 - Address validation isn't working for me: https://github.com/PayButton/paybutton/issues/18. If I enter `bitcoincash:asqwefsadfsad`, it still tries to use it. 
image


Tested on `http://localhost:10001/`.

**Pull request merged**